webui/react/src/components/MetricSelectFilter.tsx

```diff
@@ -62,16 +98,89 @@ const MetricSelectFilter: React.FC<Props> = ({ metricNames, multiple, onChange,
     (onChange as MultipleHandler)(newMetric.sort(metricNameSorter));
   }, [ multiple, onChange, value ]);
 
+  const handleFiltering = useCallback((search: string, option) => {
+    if (option.key === allOptionId || option.key === resetOptionId) {
+      return true;
+    }
+    if (!option.value) {
+      // Handle optionGroups that don't have a value to make TS happy. They aren't
+      // impacted by filtering anyway
```

Comment:
**nit:** For multi-line comments we use the following format to make them more prominent.
```
/*
 * line 1
 * line 2
 */
```
Used to have a linter do this but it made it difficult to comment and uncomment code.